File: backend/src/main/java/ch/zuehlke/fullstack/hackathon/model/Category.java
package ch.zuehlke.fullstack.hackathon.model;

import lombok.Data;

import javax.persistence.*;
import java.util.Collection;

@Entity
@Data
public class Category {
    @Id
    @GeneratedValue
    private Long categoryID;

    private String name;

    @OneToMany(mappedBy = "category", fetch = FetchType.LAZY)
    private Collection<Interest> interests;
}

File: frontend/src/app/shared/models/category.ts
import {Interest} from "@base/shared/models/interest";

export class Category {
  private readonly _id: number;
  private readonly _name: string;
  private readonly _interests: Interest[];

  constructor(id: number, name: string, interests: Interest[]) {
    this._id = id;
    this._interests = interests;
    this._name = name;
  }

  get id(): number {
    return this._id;
  }

  get name(): string {
    return this._name;
  }

  get interests(): Interest[] {
    return this._interests;
  }
}
